pkg/api/types.go

```diff
@@ -180,6 +180,139 @@ type VolumeSource struct {
 	GitRepo *GitRepoVolumeSource `json:"gitRepo"`
 	// Secret represents a secret that should populate this volume.
 	Secret *SecretVolumeSource `json:"secret"`
+	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace
+	PersistentVolumeClaimVolumeSource *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
+}
+
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+type PersistentVolumeSource struct {
+
+	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
+	// kubelet's host machine and then exposed to the pod.
+	GCEPersistentDisk *GCEPersistentDiskVolumeSource `json:"persistentDisk"`
+
+	// AWSElasticBlockStore represents an EBS volume in AWS that is attached to
+	// kubelet's host machine and then exposed to the pod.
+	AWSElasticBlockStore *AWSElasticBlockStore `json:"elasticBlockStore"`
+
+	// NFSMount represents the location of an exported volume on a fileserver
+	NFSMount *NFSMount `json:"nfsMount"`
+
+	// HostPath represents a directory on the host.  It is useful for testing PersistentVolumes.
+	HostPath *HostPathVolumeSource `json:"hostPath"`
```

Comment:
It's wonderfully convenient to test with.  Perhaps, too, for users to get acquainted with using PVs.

I will update the comment to explicitly state there is no on-host persistent storage that spans hosts, although it would work in a single node cluster or where the hostPath is available on all hosts.